lib/maths/common/COneOfNPrior.cc

```diff
@@ -320,7 +320,7 @@ void COneOfNPrior::addSamples(const TDouble1Vec& samples,
     for (std::size_t i = 0; i < samples.size(); ++i) {
         double xi = samples[i];
         double ni = maths_t::countForUpdate(weights[i]);
```

Comment:
nit: calculation of `ni` could move inside the `if`

Reply:
Done in [bc63e9f](https://github.com/elastic/ml-cpp/pull/2197/commits/bc63e9fa078fe98f7386f8e5c3b995a59dd50f0b).